contracts/SampleVerifier.sol

```diff
@@ -0,0 +1,253 @@
+pragma solidity 0.4.25;
+pragma experimental ABIEncoderV2;
+
+import "./IEnforcer.sol";
+import "./IVerifier.sol";
+import "./IEthereumRuntime.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "./MerkleProof.slb";
+import "./Hash.slb";
+
+
+// solhint-disable-next-line max-states-count
+contract SampleVerifier is Ownable, IVerifier {
+    struct StateHash {
+        bytes32 hash;
+        uint256 step;
+    }
+
+    struct ComputationHash {
+        bytes32 merkleRoot;
+        uint256 stepCount;
+    }
+
+    struct Dispute {
+        bytes32 executionId;
+        address solver;
+        address challenger;
```

Comment:
can we not remove the authentication (solver and challenger address) here?
instead use `solverComputationHash` and `challengerComputationHash` to authenticate parties?

Reply:
i mean that in `solverProofs()` the correct proof would be enough. we would know that the party has the same computation path.

Reply:
OK, this makes sense
but I think `solver` and `challenger` is still needed in `respond` :thinking:

Reply:
solver address can be removed, but challenger address is needed in enforcer's result function. should we keep it? or enforcer has to store challenger addres

Reply:
I think the Enforcer contract should keep track of solver- & challenger addrs via the DisputeId.
@johannbarbie What's your take on this?

Reply:
@johannbarbie @pinkiebell 
I have remove `solver` address from Dispute